### ISSUE TITLE
Create APPX installer

### ISSUE THREAD

**Issue by ridomin** [tannerhelland/PhotoDemon#251]
Create an APPX installer for this application, it will benefit from AppContainers features, and will allow store submission.

**Comment by Kroc** [comment]
VB6 programs are not allowed on the store! It's specifically mentioned by Microsoft.

**Comment by tannerhelland** [comment]
Hi @ridomin.  Thank you for the suggestion.

I'm afraid I do not have any experience with creating APPX installers, nor do I have access to a code-signing certificate that would work for PhotoDemon.  Given the costs involved in obtaining such certificates, and the rate at which donations roll in for a project like this one... it will be a very long time before I have the cash to afford a code-signing certificate for PD.

I don't know the ins and outs of the APPX process, especially for a VB6 app, but if you have any resources on the topic, I'll happily take a look at them.

**Comment by ridomin** [comment]
@Kroc I dont think so, there are some [posts ](https://blogs.windows.com/buildingapps/2017/02/01/adding-uwp-features-existing-pc-software) covering VB6 with APPX.

@tannerhelland  you don't need to buy any certificate, all you need is a dev center account ($19) and request permission to submit the app to the store using this [form](https://developer.microsoft.com/en-us/windows/projects/campaigns/desktop-bridge).  

Let me know if you have any issues.

**Comment by ridomin** [comment]
How can I help you to publish to the Store?

**Comment by vatterspun** [comment]
Certainly there are a lot of Github tickets on this topic and I'm not as active on this topic as I once was, but for those freeware/OSS devs, none have posted their work to the MS app store.

Certainly some systems are locked down to only those from the app store, and users can presumably be more sure that they're safe and virus-free.  However, I don't know how common this is nor would I guess about how many users would try PhotoDemon because it's in the app store.  Microsoft dumping MS Photo Editor for Office 2016 might be a nice opening, but that's just a guess.

**Comment by ridomin** [comment]
It's hard to tell how many users will try PhotoDaemon if it's in the store. But there are a number of benefits: For your users, they can get updates automatically, for you as developer/author you will get analytics, feedback etc..

I can help you to produce the APPX package and publish to the store, or help you to get the Store account.

**Comment by tannerhelland** [comment]
Hi @ridomin.  Sorry that I have not updated this issue for some time.  My wife gave birth last week, and I have not had time for PhotoDemon due to ongoing family duties.  I hope to resume regular work on it later this year.

To be honest, store submission will not be high on my priority list until a new stable release is ready.  (Otherwise, I'd be wasting time submitting an out-of-date stable build to the store.)  In addition, many of the store "features" you describe (like automatic updates) are already implemented in PhotoDemon, which is another reason I have not pursued this more aggressively.

Once a new stable build has entered the beta phase, let's revisit this.  Until then, I'd like to focus my energy on improving PhotoDemon's feature set, particularly in areas where it is not currently competitive (such as lacking any kind of shape tool).  

Thank you for continuing to remind me of this, however - and if a few months go by without me saying anything, feel free to pop in again!  :)

**Comment by ridomin** [comment]
Thanks for the reply. If you have an stable version I can help you with package creation and even store submission, but I guess you would prefer to keep it under you account. If you are interested I can ask for a promotional code so you dont need to pay anything. Just let me know what is more convenient to you.

**Comment by vatterspun** [comment]
> My wife gave birth last week, and I have not had time for PhotoDemon due to ongoing family duties.

That's fantastic!  Awesome news :)

**Comment by jpbro** [comment]
Congratulations @tannerhelland! Best wishes to you and your family :)

**Comment by vatterspun** [comment]
> To be honest, store submission will not be high on my priority list until a new stable release is ready. 

I'm very unhappy with the Microsoft Store and how it's treated some other excellent projects listed here on Github, so certainly feel free to delay longer: https://github.com/MathewSachin/Captura/issues/414

**Comment by tannerhelland** [comment]
Thank you for the update, @vatterspun.  

I hadn't considered searching Microsoft's Store for PhotoDemon rip-offs.  I decided to give it a try, and it took me less than three minutes to find a full rip of PhotoDemon on the store (under the name "Photo Magick"; I won't link it directly, but it's an obvious rip of a random PD nightly build, right down to the project description and screenshots).  

It is discouraging that MS is so incredibly lax with their vetting procedures.

I've reported the app to MS but haven't heard anything back yet.  Needless to say, this doesn't exactly encourage me to participate in their app ecosystem.  :(

**Comment by vatterspun** [comment]
> It is discouraging that MS is so incredibly lax with their vetting procedures.

Agreed 100% 

> I've reported the app to MS but haven't heard anything back yet. Needless to say, this doesn't exactly encourage me to participate in their app ecosystem. :(

Based on my experience so far, you should expect it to stay there.  Softpedia meanwhile got it resolved almost immediately.

**Comment by tannerhelland** [comment]
Well, this sucks - nearly a month later, and no response whatsoever from Microsoft.  Glad to see they care so much about the contents of their store... /s

Because I'm seeing this issue linked elsewhere, I want to clarify something: most open-source licenses (including this project's BSD license, MIT, GPL/LGPL, etc) do not prohibit 3rd-parties from reselling open-source software.  That may be an ethical issue, but it's *not* a legal one, and it's not something most open-source licenses attempt to prevent.  If you think you can resell free software for a price, more power to you.

The issue is that these Microsoft Store clones are removing all copyright and license information from the open-source projects they've copied.  In PhotoDemon's case, the clone stripped not only PhotoDemon's BSD license, it's also stripped all 3rd-party licenses shipped with the project.  These licenses are additionally linked from PhotoDemon's Help menu, but the clone removed that menu (presumably in an attempt to hide the program's true source).

**That's** the real issue here, and [it *is* legally actionable](https://digitalbusiness.law/2017/01/enforcement-of-open-source-licenses/). 

Microsoft is insane to ignore this, and I'll pursue a stronger recourse as time allows.  In the meantime, I hope people continue to draw attention to this issue, because it's only going to get worse unless people force Microsoft to act.

**Comment by vatterspun** [comment]
Thank you for your input.  I agree that this is terrible PR and contradictory to so many other messages MS is trying to send to the market.

Related: https://www.ghacks.net/2019/06/17/fake-commercial-versions-of-open-source-applications-on-the-microsoft-store/  Please feel free to reach out to Martin of ghacks directly and to add emphasis that developers are feeling chilled https://www.ghacks.net/contact/ .  I think he'll publish it.
